### DISCUSSION TITLE
Use dolphin instead of nautilus in firefox

### DISCUSSION THREAD

**Discussion by Gigas002** [prasanthrangan/hyprdots#1017] [Q&A] [answered]
As title says, is there a way for it? On KDE Plasma session, I have this stuff in my `user.js`, which should force usage of dolphin:

```js
user_pref("widget.use-xdg-desktop-portal", true);
user_pref("widget.use-xdg-desktop-portal.kde", true);
user_pref("GTK_USE_PORTAL", 1);
user_pref("widget.use-xdg-desktop-portal.file-picker", 1);
```

However, firefox still uses nautilus.
I suspect, that *might* be out of scope for this project, though. I remember reading something like, `xdg-desktop-portal-hyprland` doesn't work well with other `xdg-desktop-portal`, which might be the cause of it, but yet again I'm not sure.

**Comment by kRHYME7** [comment]
I hope you can find answers for this too. I have the same issue. I installed nautilus just to have the file manager pop out. 

I already set xdg-mime but not recognized. 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/88a4cff0-3338-4329-98f4-a10f0291e913)

**Reply by Gigas002** [reply]
Alright, I'll text the update here if I'll know anything else then :+1:

**Reply by Gigas002** [reply]
Same thing happens to `code`/`codium` btw. Works on plasma, but not on `hyprland` session, so I still think it's `xdg-portal` to blame here.

For a reference, found two links on arch wiki, which *may* help somehow, though results on my system are not promising:

- https://wiki.archlinux.org/title/Firefox#XDG_Desktop_Portal_integration
- https://wiki.archlinux.org/title/Uniform_look_for_Qt_and_GTK_applications#Consistent_file_dialog_under_KDE_Plasma

I find the note about `GDK_DEBUG=portals` especially interesting, didn't hear about it before.

**Reply by kRHYME7** [reply]
I'll read it later, but I think someone already give an answer to me on this matter. I'll try to find it. 

Also have you tried using another browser any chromium base like brave. What is its behavior?

**Reply by Gigas002** [reply]
Setting the `GDK_DEBUG`/`GTK_USE_PORTAL` takes no effect (testing with `thorium`). I don't know if there are any specific flags for it in chromium browsers, but I tried playing with `[#gtk-version](chrome://flags/#gtk-version)` flag, no effect whatsoever.

**Reply by Gigas002** [reply]
I think we may try to trick specific apps that these are running from kde plasma, not showing real WM somehow: https://wiki.archlinux.org/title/XDG_Desktop_Portal
(setting the `XDG_CURRENT_DESKTOP=KDE` didn't took any result though)

**Reply by Gigas002** [reply, marked as answer]
@kRHYME7 I managed to get it working. Looks ugly af (I think some `windowrule`s might do a trick here?), but at least working.
For it to be working, I've created the `~/.config/xdg-desktop-portal` directory with `hyprland-portals.conf` file. File's contents:

```ini
[preferred]
default=hyprland;gtk
org.freedesktop.impl.portal.FileChooser=kde
```

The only addition between default config (located in `/usr/share/xdg-desktop-portal/hyprland-portals.conf`) is the last line which makes it work out.

I think you also need to restart some systemd service in order for it to work with these changes, but I simply rebooted the pc.

**Reply by kRHYME7** [reply]
@Gigas002 Let me try this tomorrow! Thank you, this is the feature I lack. I hope this works on my end. 


 **Update** I can't get it working. 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/cdb93683-ec98-438d-abd4-08170e308d93)


Originally, I do not have this file. I only have the one with ".bak" as suffix.
```
/usr/share/xdg-desktop-portal/hyprland-portals.conf
```

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/cdb93683-ec98-438d-abd4-08170e308d93)

**Reply by Gigas002** [reply]
@kRHYME7 Hmm. You do have the `xdg-desktop-portal`, `xdg-desktop-portal-hyprland` and `xdg-desktop-portal-kde` packages installed, right?
Can you please also try changing the `default` value to `default=hyprland;kde`? That worked on my end too

**Reply by kRHYME7** [reply]
I installed then remove the KDE portal, and it now works.  I tried removing all the files I modified, and it still works. LOL But nah at least it works now

**Reply by Gigas002** [reply]
Going to take a look at `qt5ct-kde` and `qt6ct-kde` packages behavior. By the looks of it, it can apply themes to portal stuff, may fix the ugliness of dialogs

**Reply by Gigas002** [reply]
Update: tried using these two packages above and set the `windowrulev2 = opacity 0.80 0.80,class:^(org.freedesktop.impl.portal.FileChooser)$` rule, no effect, the dialog's still ugly

**Reply by Gigas002** [reply]
A-ha! Here's the solution:

```
windowrulev2 = opacity 0.80 0.80,class:^(org.freedesktop.impl.portal.desktop.gtk)$
windowrulev2 = opacity 0.80 0.80,class:^(org.freedesktop.impl.portal.desktop.kde)$
windowrulev2 = opacity 0.80 0.80,class:^(org.freedesktop.impl.portal.desktop.hyprland)$
```

**Reply by seyfu-t** [reply]
> I installed then remove the KDE portal, and it now works. I tried removing all the files I modified, and it still works. LOL But nah at least it works now

I am at this point but re-installing does nothing... did you do something else after re-installing?
